### PR TITLE
fix(api): revamp reorged slots handling logic

### PR DESCRIPTION
**Pull request by PJColombo**
Revamped the reorged slots handling endpoint.

The endpoint now accepts a set of forked slots and marks those that exist in the database as reorganized. This is achieved by transferring the transactions of the corresponding blocks to the fork transactions table.

### CODE REVIEW

File: packages/api/src/routers/indexer/handleReorgedSlots.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      )
+    );
+
+    let totalUpdatedSlots = 0;
+
+    if (typeof result === "number") {
+      totalUpdatedSlots = result;
+    } else {
+      totalUpdatedSlots = result[0]?.count ?? 0;
+    }
```
[the remark this answers is not in the document]
Comment (0xGabi) [reply packages/api/src/routers/indexer/handleReorgedSlots.ts:59]: I think in this case we can be sure that the result is always number. Right?

File: packages/api/test/indexer.test.ts [hunk 2]
```diff
@@ -527,18 +527,51 @@ describe("Indexer router", async () => {
...
 
-        const transactionForks =
-          await authorizedContext.prisma.transactionFork.findMany();
+        expect(result).toEqual({
+          totalUpdatedSlots: input.reorgedSlots.length,
+        });
+      });
+
+      it("should ignore non-existent slots and mark the ones that exist as reorged", async () => {
+        const reorgedSlots = [106, 107, 99999];
```
Comment (0xGabi) [inline packages/api/test/indexer.test.ts:539]: Maybe we can also test a request of non-existent slots alone as well?

**Review by 0xGabi — APPROVED** [review]
Looking good!

File: packages/api/src/routers/indexer/handleReorgedSlots.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (PJColombo) [reply packages/api/src/routers/indexer/handleReorgedSlots.ts:59]: Yes, this was done just to comply with ts type enforcement but we perform a cast as well

File: packages/api/test/indexer.test.ts (hunk 2 above)
Comment (PJColombo) [reply packages/api/test/indexer.test.ts:539]: Sure

**Comment by Mandog1120** [comment]
e77bb5cfa800f059f5d5310f311435120580553d

**Pull request merged**